interfaces/Portalicious/src/app/services/auth.service.ts

```diff
@@ -153,4 +154,50 @@ export class AuthService {
       );
     }
   }
+
+  public async refreshCurrentUser() {
+    const userDto = await this.apiService.getCurrentUser();
+
+    if (!userDto?.user) {
+      await this.logout();
+      return;
+    }
+
+    this.setUserInStorage(userDto.user);
+  }
+
+  private isAssignedToProgram(
+    programId: number,
+    user?: LocalStorageUser | null,
+  ): boolean {
+    user = user ?? this.user;
+    return (
+      !!user?.permissions &&
+      Object.keys(user.permissions).includes(String(programId))
+    );
+  }
+
+  public hasPermission(
+    programId: number,
+    requiredPermission: PermissionEnum,
+    user?: LocalStorageUser | null,
+  ): boolean {
+    user = user ?? this.user;
+    // During development: Use this to simulate a user not having a certain permission
+    // user.permissions[programId] = user.permissions[programId].filter(
+    //   (p) => p !== Permission.FspDebitCardBLOCK,
+    // );
+
+    // TODO: Move this to a better place in the flow, so it doesn't have to be checked this often
+    // Check with Azure (again) when user has no permissions (yet)
+    if (!this.isAssignedToProgram(programId, user)) {
+      void this.refreshCurrentUser(); // Don't await, as it will block all permission-checks
+    }
```

Comment:
😭😭😭

I hope we can do without taking this skeleton-from-the-old-portal with us.. 🤞

Reply:
Tbh I'm not entirely sure what is going on here. I blindly brought it over but we can decide not to unless we really have to.

Reply:
I think we can do without this (for now) and reconsider this while we work on the "add SSO-login"-features...

I'd like to limit all the introduced workarounds/edge-cases we start with.